components/TimeSeries.vue

```diff
@@ -0,0 +1,353 @@
+ */
+const syncTooltipsAndCrosshairs = throttle(() => {
+  if (chart.hoverPoint) {
+    allCharts.value.forEach(({ series }) => {
+      // Get the point with the same x as the hovered point
+      const point = series[0].getValidPoints().find(({ x }) => x === chart.hoverPoint!.x);
+
+      if (point && point !== chart.hoverPoint) {
+        point.onMouseOver();
+      }
+    });
+  };
+}, 100, { leading: true });
+
+const onMove = () => syncTooltipsAndCrosshairs();
+
+// Override the reset function as per synchronisation demo: https://www.highcharts.com/demo/highcharts/synchronized-charts
+Highcharts.Pointer.prototype.reset = () => {
+  return undefined;
+};
+
+// Synchronize zooming through the setExtremes event handler.
+const syncExtremes = (event: { trigger: string, min: number | undefined, max: number | undefined }) => {
+  if (event.trigger !== "syncExtremes") {
+    allCharts.value.forEach((ch: Highcharts.Chart) => {
+      if (ch.xAxis[0].setExtremes) { // setExtremes is null while updating
+        ch.xAxis[0].setExtremes(event.min, event.max, undefined, { duration: 250 }, { trigger: "syncExtremes" });
+        if (event.min && event.max) {
+          // Prevent exporting while zoomed in since image export doesn't work well when zoomed in
+          ch.update({ chart: { className: undefined }, exporting: { enabled: false } });
+        } else {
+          ch.update({ chart: { className: hideResetZoomButtonClassName }, exporting: { enabled: true } });
+        }
+      }
+    });
+  }
+};
+
+const chartInitialOptions = () => {
+  return {
+    chart: {
+      height: chartHeight(),
+      marginLeft: 75, // Specify the margin of the y-axis so that all charts' left edges are lined up
+      backgroundColor: chartBackgroundColor,
+      events: {
+        fullscreenOpen() {
+          this.update({ chart: { backgroundColor: chartBackgroundColorOnExporting } });
+        },
+        fullscreenClose() {
+          this.update({ chart: { backgroundColor: chartBackgroundColor } });
+        },
+      },
+      className: hideResetZoomButtonClassName,
+      style: {
+        fontFamily: "ImperialSansText, sans-serif", // TODO: Make the font-family derive from a globally configurable constant
+      },
+      zooming: {
+        type: "x",
+      },
+    },
+    exporting: {
+      filename: `${seriesMetadata.value!.label} in ${appStore.currentScenario.parameters?.country}`,
+      chartOptions: {
+        title: {
+          text: seriesMetadata.value!.label,
+        },
+        subtitle: {
+          text: seriesMetadata.value!.description,
+        },
+        chart: {
+          backgroundColor: chartBackgroundColorOnExporting,
+          height: 500,
+        },
+      },
+      buttons: {
+        contextButton: {
+          height: 20,
+          width: 22,
+          symbolSize: 12,
+          symbolY: 10,
+          symbolX: 12,
+          symbolStrokeWidth: 2,
+          // Omit 'printChart' and 'viewData' from menu items
```

Comment:
I guess `viewData` is limited usefulness but someone might want to print..?

Reply:
I omitted printChart because whenever you print, there is a mouseover, which induces a tooltip, which is included in the print preview, which is bad, and likely not worth the effort to fix.

![image](https://github.com/user-attachments/assets/b721a750-aac0-4bc8-902c-6ac06e7908b5)

Reply:
Downloading the PDF, PNG, JPEG, or SVG should suffice for any printing needs!